### ISSUE TITLE
BUG: depending on the platform , the font layout changes

### ISSUE THREAD

**Issue by anatoliibakhtin** [zsviczian/obsidian-excalidraw-plugin#1001]
Hello dear developer, my problem:

### **Environment**

| PHONE SYSTEM INFO:                               | PC SYSTEM INFO:                                  |
| ------------------------------------------------ | ------------------------------------------------ |
| Operating system: android 9 (samsung SM-G892U)   |Obsidian version: v1.1.9                         |
| Obsidian version: 1.4.1 (83)                     |Installer version: v1.1.9                        |
| API version: v1.0.3                              |Operating system: Windows 10 Pro 10.0.22000      |
| Login status: not logged in                      |Login status: not logged in                      |
| |Insider build toggle: off                        |
| Live preview: on                                 |Live preview: on                                 |
| Legacy editor: off                               |Legacy editor: off                               |
| Base theme: light                                |Base theme: light                                |
| Community theme: none                            |Community theme: none                            |
| Snippets enabled: 0                              |Snippets enabled: 0                              |
| Restricted mode: off                             |Restricted mode: off                             |
| Plugins installed: 43                            |Plugins installed: 43                            |
| Plugins enabled: 19                              |Plugins enabled: 23                              |
| 1: Tag Wrangler v0.5.6                           |1: Tag Wrangler v0.5.6                           |
| 2: Excalidraw v1.8.12                            |2: Excalidraw v1.8.12                            |
| 3: Smart Random Note v0.2.1                      |3: Smart Random Note v0.2.1                      |
| 4: Hotkeys++ v0.2.7                              |4: Hotkeys++ v0.2.7                              |
| 5: Customizable Page Header and Title Bar v4.6.2 |5: Customizable Page Header and Title Bar v4.6.2 |
| 6: Outliner v4.2.1                               |6: Outliner v4.2.1                               |
| 7: Custom Frames v2.4.4                          |7: Custom Frames v2.4.4                          |
| 8: Dataview v0.5.47                              |8: Dataview v0.5.47                              |
| 9: Recent Files v1.3.5                           |9: Recent Files v1.3.5                           |
| 10: Hotkey Helper v0.3.15                        |10: Hotkey Helper v0.3.15                        |
| 11: Note Refactor v1.7.1                         |11: Pane Relief v0.4.1                           |
| 12: Hover Editor v0.11.9                         |12: Note Refactor v1.7.1                         |
| 13: Quick Switcher++ v2.3.5                      |13: Hover Editor v0.11.9                         |
| 14: Advanced URI v1.31.2                         |14: Quick Switcher++ v2.3.5                      |
| 15: ExcaliBrain v0.1.11                          |15: Advanced URI v1.31.2                         |
| 16: LanguageTool Integration v0.3.3              |16: ExcaliBrain v0.1.11                          |
| 17: Minimal Theme Settings v6.2.0                |17: Local images v0.14.2                         |
| 18: Tasks v1.22.0                                |18: LanguageTool Integration v0.3.3              |
| 19: Omnisearch v1.11.0                           |19: Core Search Assistant v0.9.1                 |
||20: Pandoc Plugin v0.4.0                         |
||21: Minimal Theme Settings v6.2.0                |
||22: Tasks v1.22.0                                |
||23: Omnisearch v1.11.0                           |

### **The bug**
Depending on the platform where I open the excalidraw file the font layout changes. Mostly family code, least of all handwritten. I tried to do the same on excalidraw.com exported the drawing from the computer version and opened it in the phone got slightly different results, but it's nothing compared to the situation in obsidian.

![image](https://user-images.githubusercontent.com/71871141/216166462-e77f48e6-ef89-4100-b99d-7269381a54e7.png)

example: obsidian.exe only

https://user-images.githubusercontent.com/71871141/216166603-ffab0d8a-e7ec-4b04-a27b-0e6fd8007983.mp4

example: excalidraw.com only

https://user-images.githubusercontent.com/71871141/216166623-64cb0867-8aa0-4732-93b2-3909e8d00092.mp4

### **To Reproduce**

I may have done a lot of unnecessary things, but I had no other way of knowing that only the location of the text would change, regardless of any parameters (grouping, blocking, how it shifts to another line by "enter" or "script" or "shape").

https://user-images.githubusercontent.com/71871141/216172239-66957c80-e4a6-4643-8c0e-b7c994cabdb3.mp4

1. create a text object with multiple lines (you can use the enter key, script, geometric shape, the more lines, i.e. the lower the font, the more the offset will be inside the container for text).
2. use ink on top of the font, or that any other element of the canvas, except the font to highlight the text you want (for accuracy in tracking the offset of the font, I painted the letters O).
3. synchronize excalidraw between your devices (copy, replace. no matter if you transfer files with plugin settings or only drawings the result will be the same)
4. look at how on the phone your font is compressed, recompressed and... it turns out your selections in the text were missed, because the font has changed its location.

### **Expected behavior**
I expect the text and all the other elements to not move or change its position and show the same result (not one pixel should change its coordinates), no matter what platform it was opened on.

### **Additional context**

The problem seemed to be in all versions of obsidian... But it is not solved, that's why I am writing. 

- [x] Can you reproduce it ?
- [x] Can it be solved by you ? 

upd: If not, not a big problem - I'll just change my workflow. It's probably better this way, since everything on the canvas is constantly moving small text to the canvas, large text to the document. But I will leave the problem open until you reply.

Thank you for your attention !🌻

**Comment by zsviczian** [comment]
Yes, this is a known issue. I've played a lot with this in the past as this bothered me a lot, but I was not able to get to a satisfactory solution. The issue has to do with device pixel ratio and default font sizes on different devices and browsers. I guess on top of this, some of these default values are modified by Obsidian in Electron, but this is just a guess, as I was not able to work out the dependencies impacting size differences.

**Comment by anatoliibakhtin** [comment]
@zsviczian, the text almost does not jump, more or less working version ).

https://github.com/zsviczian/obsidian-excalidraw-plugin/assets/71871141/f84c2795-b960-4ae1-8a88-cd1be7526885

**Comment by anatoliibakhtin** [comment]
I don't see that problem anymore
